sdks/java/core/src/test/java/org/apache/beam/sdk/testing/InterceptingUrlClassLoader.java

```diff
@@ -17,22 +17,32 @@
  */
 package org.apache.beam.sdk.testing;
 
+import com.google.common.base.Predicates;
 import com.google.common.collect.Sets;
 import com.google.common.io.ByteStreams;
 import java.io.IOException;
-import java.util.Set;
+import java.util.function.Predicate;
 
 /**
  * A classloader that intercepts loading of specifically named classes. This classloader copies
  * the original classes definition and is useful for testing code which needs to validate usage
  * with multiple classloaders..
  */
 public class InterceptingUrlClassLoader extends ClassLoader {
-    private final Set<String> ownedClasses;
+    private final Predicate<String> test;
+
+    private InterceptingUrlClassLoader(final ClassLoader parent,
+                                       final com.google.common.base.Predicate<String> predicate) {
+        this(parent, (Predicate<String>) predicate::apply);
+    }
 
     public InterceptingUrlClassLoader(final ClassLoader parent, final String... ownedClasses) {
+        this(parent, Predicates.in(Sets.newHashSet(ownedClasses)));
```

Comment:
You don't need to have the private constructor for the guava predicate if you just pass in the method reference to `apply` here.